### PR TITLE
Empty string for the config setting for proxies did not really work

### PR DESCRIPTION
**Pull request by smurawski**
/cc @cheeseplus

### CODE REVIEW

**Comment by jjasghar** [comment]
💥 goes the dyno-mite!

**Comment by smurawski** [comment]
The problem with the PR we merged a while back is

```
!config[:http_proxy].empty?
```

since 

```
"".empty? == true
```

so, it doesn't see the empty setting for the proxy and defaults to using the environmental variables.

File: lib/kitchen/configurable.rb [hunk 1]
```diff
@@ -306,28 +306,28 @@ def validate_config!
...
+        env.join("\n").concat("\n").concat(code)
       else
-        export_proxy(env, "ftp")
+        Util.wrap_command(env.join("\n").concat("\n").concat(code))
       end
+    end
+
+    # rubocop:disable Metrics/AbcSize, Metrics/PerceivedComplexity, Metrics/CyclomaticComplexity
+    def proxy_settings
+      env = [:http_proxy, :https_proxy, :ftp_proxy].collect do |protocol|
```
Comment (mwrock) [inline lib/kitchen/configurable.rb]: you could do:
```
env = [:http_proxy, :https_proxy, :ftp_proxy].each_with_object([]) do |protocol, set_env|
        if !config.key?(protocol) || config[protocol].nil?
          export_proxy(set_env, protocol)
        elsif proxy_config_setting_present?(protocol)
          set_env << shell_env_var("#{protocol.downcase}", config[protocol])
          set_env << shell_env_var("#{protocol.upcase}", config[protocol])
        end
      end
```
and that may allow you to get rid of the rubocop suppression.

File: lib/kitchen/configurable.rb [hunk 2]
```diff
@@ -306,28 +306,28 @@ def validate_config!
...
-      if config[:https_proxy] && !config[:https_proxy].empty?
-        env << shell_env_var("https_proxy", config[:https_proxy])
-        env << shell_env_var("HTTPS_PROXY", config[:https_proxy])
-      else
-        export_proxy(env, "https")
-      end
-      if config[:ftp_proxy] && !config[:ftp_proxy].empty?
-        env << shell_env_var("ftp_proxy", config[:ftp_proxy])
-        env << shell_env_var("FTP_PROXY", config[:ftp_proxy])
+      env = proxy_settings
```
Comment (mwrock) [inline lib/kitchen/configurable.rb]: try `env = (proxy_settings << code).join(\n)`

**Comment by mwrock** [comment]
Im :+1: with or without my suggestions

**Comment by cheeseplus** [comment]
👍

**Pull request merged**